### ISSUE TITLE
Coding Standards

### ISSUE THREAD

**Issue by netniV** [howardjones/network-weathermap#159]
As more and more people look to contribute their additions to your code, you may want to define a coding standard.  Just looking at the WeatherMapCactiManagementPlugin.php file, I noticed several differing styles of coding for function definitions.  I know your wiki has mention of using PSR2 briefly, but often it's worth having a page that spells things out.

**Comment by howardjones** [comment]
There is this, but it doesn't (yet) mention coding standards. I mostly follow PSR1/PSR2, except for the line length restriction. It would indeed be good to write one up. The phpcs rules I use are in the build/ directory.

**Comment by howardjones** [comment]
The other one, especially for core features (not so much for UI, since I haven't figured out a nice way), is to try to add a test for a new bug, as well as the fix (e.g. Alex's NOTES bug the other day)

**Comment by netniV** [comment]
> There is this, but it doesn't (yet) mention coding standards. I mostly follow PSR1/PSR2, except for the line length restriction. It would indeed be good to write one up. The phpcs rules I use are in the root directory.

Not quite sure what "this" is you are referring to :) 

Some examples of the different styles:

```
public function makeURL($params, $altURL = "")
{
```
```
    public
    function handleDumpMapsAPI(
        $request,
        $appObject
    ) {
```

I can submit .a PR to correct these, or would you rather I just left it with you?

**Comment by howardjones** [comment]
D'oh. I didn't paste the link :-)

THIS https://github.com/howardjones/network-weathermap/blob/master/dev/dev-README.md

**Comment by howardjones** [comment]
Having run it again for the first time in a while, the other thing is that it'll never run clean in any of the Cacti-related code, because Cacti doesn't use PSR2, so there will always be a bunch of:

` 202 | ERROR | [ ] Variable "user_auth_realm_filenames" is not in valid camel caps format`

for things defined in Cacti's `config.h`.  This isn't a problem, just an observation, and a reason not to use phpcs as any kind of formal requirement for new code.

**Comment by netniV** [comment]
True.  The Cacti Group code does seem to be based around using underscores for the most part.  You would have to be selective as to which rules to apply I guess.  We have had to do a similar thing for Markdown with the documentation we have done recently.  It flags an error with the markdown lint if you jump more than one level for a subtitle but we use `######` for captions of tables/images so they can be directly linkable.

**Comment by howardjones** [comment]
I'm pretty sure I can exclude particular rules per file, or just exclude the Integrations/Cacti directory from the normal phpcs pass. I've just run into another fun one, where one of the phpcs standards I used are now marked as requiring PHP 7.1 or above in composer, which is mostly not a problem for a dev environment, but means that only half of the Travis builds work. Having Travis test with various PHP versions is very useful, so goodbye `slevomat/coding-standards`!

**Comment by netniV** [comment]
Always fun.  You could always fork and make your own?
